lnwire/init_message.go

```diff
@@ -52,11 +57,31 @@ var _ SizeableMessage = (*Init)(nil)
 //
 // This is part of the lnwire.Message interface.
 func (msg *Init) Decode(r io.Reader, pver uint32) error {
-	return ReadElements(r,
+	err := ReadElements(r,
 		&msg.GlobalFeatures,
 		&msg.Features,
 		&msg.ExtraData,
 	)
+	if err != nil {
+		return err
+	}
+
+	customRecords, _, extraData, err := ParseAndExtractCustomRecords(
+		msg.ExtraData,
+	)
+	if err != nil {
+		return err
+	}
+
+	if len(customRecords) > 0 {
+		msg.CustomRecords = customRecords
+	}
+
+	if len(extraData) > 0 {
+		msg.ExtraData = extraData
```

Comment:
This means the `msg.ExtraData` will always have the raw custom data, when the data read from the wire only contains custom records. I think we should add tests here, similar to https://github.com/lightningnetwork/lnd/blob/master/lnwire/dyn_commit_test.go or https://github.com/lightningnetwork/lnd/blob/master/lnwire/channel_update_2_test.go

There's also #10140 that can be used as example.